client/proposer/service_test.go

```diff
@@ -47,19 +53,47 @@ func settingUpProposer(t *testing.T) (*Proposer, *internal.MockClient) {
 		t.Fatalf("Failed to start syncer %v", err)
 	}
 
-	fakeProposer, err := NewProposer(params.DefaultConfig, node, server, pool, db, 1, fakeSyncer)
+	c := params.GetDefaultConfig()
+	fakeProposer, err := NewProposer(c, node, server, pool, db, 1, fakeSyncer)
 	if err != nil {
 		t.Fatalf("Failed to create proposer %v", err)
 	}
 	fakeProposer.config.CollationSizeLimit = int64(math.Pow(float64(2), float64(10)))
 
 	return fakeProposer, node
+}
 
+// waitForLogMsg keeps checking for log messages until "want" is found or a
+// timeout expires, returning the index of the message if found.
+// The motivation is to be more resilient to connection failure errors that can
+// appear in the log:
+// Failed to connect to peer: dial attempt failed: <peer.ID 16Uiu2> --> <peer.ID 16Uiu2> dial attempt failed: connection refused
+func waitForLogMsg(hook *logTest.Hook, want string) (int, error) {
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		case <-timeout:
+			return 0, fmt.Errorf("didn't get %q log entry on time", want)
+		case <-tick:
+			for index, msg := range hook.AllEntries() {
+				if msg.Message == want {
+					return index, nil
+				}
+			}
+		}
+	}
 }
 
 func TestProposerRoundTrip(t *testing.T) {
 	hook := logTest.NewGlobal()
 	fakeProposer, node := settingUpProposer(t)
+	defer func() {
+		fakeProposer.Stop()
```

Comment:
One more thing for all the tests can you add `defer fakeProposer.Stop()` below `fakeProposer.Start()` instead of having it at the top. Just to make it clear that the service is being handled properly

Reply:
Sure, done.